iothub/service/src/IotHubTokenCredential.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using System.Threading;
+
+#if !NET451
+
+using Azure.Core;
+
+#endif
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// The properties required for authentication to IoT hub using a token credential.
+    /// </summary>
+    internal class IotHubTokenCredential : IotHubCredential
```

Comment:
The new TokenCredential class. Will be adding AzureSasCredential in a subsequent PR.